airflow/models/dag.py

```diff
@@ -1368,6 +1370,7 @@ def _get_task_instances(
         self,
         *,
         task_ids,
+        task_ids_and_map_indexes,
```

Comment:
Maybe merge these like `set_states`? (same for other arguments below, including the `exclude` ones)

Also maybe add type annotations to these, it’s not easy to tell what should be passed into these at first sight.